src/Middleware/OutputCaching/test/OutputCacheMiddlewareTests.cs

```diff
@@ -826,6 +824,53 @@ public async Task Locking_PreventsConcurrentRequests()
         Assert.Equal(1, responseCounter);
     }
 
+    [Fact]
+    public async Task Locking_IgnoresNonCacheableResponses()
+    {
+        var responseCounter = 0;
+
+        var task1Executing = new ManualResetEventSlim(false);
```

Comment:
Use `TaskCompletionSource`

Reply:
It's funny because I could swear I wrote a similar unit test a few years ago and you made the same comment ... I can't learn

Reply:
Updated